corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py

```diff
@@ -107,7 +116,20 @@ def migrate_from_migration(cls, apps, schema_editor):
     def couch_db(cls):
         return couch_config.get_db(cls.couch_db_slug())
 
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--verify',
+            action='store_true',
+            dest='verify',
+            default=False,
+            help="""
+                Don't migrate anything, instead check if couch and sql data is identical.
+                Only works for migrations that use a couch_id on the sql model.
```

Comment:
Would this have caught the issue before it occurred? I thought the issue was caused by a double-save, which resulted in bad state, and by the time the second save finished the damage was done. Is this intended to be run immediately after a migration to provide an early warning if things have gone badly? If yes, would it make sense to incorporate the verification into the migration in a way that each object is verified immediately after it is migrated, and if verification fails then abort the migration immediately before migrating anything else?

Reply:
It's intended to be run immediately after a migration, yes. I really like the idea of also verifying during the migration and failing if there are problems, added a couple of commits to do that.